docs/src/2.4.ManagingCamera.mdx

```diff
@@ -0,0 +1,168 @@
+
+```js
+// src/utils/useRequestAnimationFrame.js
+import { useEffect, useRef } from "react";
+
+export default function useRequestAnimationFrame(callback) {
+  const callbackRef = useRef();
+  useEffect(() => {
+    callbackRef.current = callback;
+  });
+
+  useEffect(() => {
+    const stop = requestAnimationFrame(() => {
+      callbackRef.current();
+    });
+
+    return function cleanup() {
+      cancelAnimationFrame(stop);
+    };
+  });
+}
+```
+
+Lastly, let's hook everything together:
+
+```js
+// src/Example.js
+// ...other imports
+import Duck from "./Duck";
+import useRequestAnimationFrame from "./utils/useRequestAnimationFrame";
+
+export default function Example() {
+  const cameraMoveSpeed = 1 / 100;
+  // use count to derive camera phi and thetaOffset values
+  const [count, setCount] = useState(0);
+  useRequestAnimationFrame(() => {
+    // update count before each browser repaint
+    const newCount = count + 2 >= steps ? 0 : count + 1;
+    setCount(newCount);
+  });
+
+  // generate cameraState from `count`
+  const cameraState = {
+    ...DEFAULT_CAMERA_STATE,
+    phi: count * cameraMoveSpeed * Math.PI,
+    thetaOffset: count * cameraMoveSpeed * 2 * Math.PI,
+  };
+
+  // ... other old code
+  return (
+    <Worldview cameraState={cameraState}>
+      <Spheres>{[marker]}</Spheres>
+      <Axes />
+      <Duck />
+    </Worldview>
+  );
+}
+```
+
+Now the camera is moving around the scene, giving us 360 view of the duck model!
+
+_Notice we've added `cameraState={cameraState}` which turned worldview into a [controlled](https://reactjs.org/docs/forms.html) component which usually requires an `onChange` handler. In our case here, since we don't need to know what's the new internal
+`cameraState` value, no `onCameraStateChange` prop is necessary._
+
+<CodeSandboxEmbed src="https://codesandbox.io/s/qx5zrko8nj" />
+
+## Follow an Object
+
+In this step, we are going to move the duck along the knot, and move the camera as if we are following the duck around.
+
+To move the duck, let's first modify the Duck component to accept children input so the duck position can be updated dynamically:
+
+```js
+// src/Duck.js
+const DEFAULT_CHILDREN = {
+  pose: {
+    position: { x: 0, y: 0, z: 0 },
+    orientation: { x: 0, y: 0, z: 0, w: 1 },
+  },
+  scale: { x: 3, y: 3, z: 3 },
+};
+
+export default function Example({ children = DEFAULT_CHILDREN }) {
+  return <GLTFScene model={duckModel}>{children}</GLTFScene>;
+}
+```
+
+Next, we are going to remove the camera setting for `thetaOffset` and `phi` so we don't feel dizzy observing the scene while the duck is moving. We can derive the duck position from the spheres over time, and use the duck position as `target` input for `cameraState`.
+
+```js
+// src/Example.js
+
+export default function Example() {
+  // ... other old code
+  // move the duck along the the knot by extract the sphere position over times
```

Comment:
double the, remove one of em

Reply:
'by extracting the sphere positions over time'

Reply:
Yes ^